### PR TITLE
Add availability annotations for renamed methods

### LINKED ISSUE #238: Add renamed annotations to Valet 4 release to ease migration
The latest release on the Github page is 3.2.8, whereas the latest Cocoapod is 4.1.0.

4.1.0 also lacks `@deprecated` or `renamed` annotations, making migration difficult.

### PR DESCRIPTION
**Pull request by idrougge**
Fixes #238

### CODE REVIEW

File: Sources/Valet/SecureEnclaveValet.swift [hunk 1]
```diff
@@ -218,6 +218,14 @@ public final class SecureEnclaveValet: NSObject {
         try migrateObjects(matching: valet.baseKeychainQuery, removeOnCompletion: removeOnCompletion)
     }
 
+    // MARK: Deprecation stubs
```
Comment (dfed) [inline Sources/Valet/SecureEnclaveValet.swift:221]:
```suggestion
    // MARK: Renamed methods
```

File: Sources/Valet/SinglePromptSecureEnclaveValet.swift [hunk 2]
```diff
@@ -246,6 +246,14 @@ public final class SinglePromptSecureEnclaveValet: NSObject {
         try migrateObjects(matching: valet.baseKeychainQuery, removeOnCompletion: removeOnCompletion)
     }
 
+    // MARK: Deprecation stubs
```
Comment (dfed) [inline Sources/Valet/SinglePromptSecureEnclaveValet.swift:249]:
```suggestion
    // MARK: Renamed methods
```

File: Sources/Valet/Valet.swift [hunk 3]
```diff
@@ -471,6 +471,14 @@ public final class Valet: NSObject {
     }
     #endif
 
+    // MARK: Deprecation stubs
```
Comment (dfed) [inline Sources/Valet/Valet.swift:474]:
```suggestion
    // MARK: Renamed methods
```

File: Sources/Valet/Valet.swift [hunk 4]
```diff
@@ -471,6 +471,14 @@ public final class Valet: NSObject {
...
+    
+    @available(*, unavailable, renamed: "setObject(_:forKey:)")
+    public func set(object: Data, forKey key: String) -> Bool { fatalError()v }
```
Comment (dfed) [inline Sources/Valet/Valet.swift:477]:
```suggestion
    public func set(object: Data, forKey key: String) -> Bool { fatalError() }
```

File: Sources/Valet/SecureEnclaveValet.swift (hunk 1 above)
Comment (NickEntin) [reply Sources/Valet/SecureEnclaveValet.swift:221]: nit: `Renamed Methods` to match capitalization of other marks

Comment (dfed) [reply Sources/Valet/SecureEnclaveValet.swift:221]: updated everywhere

**Pull request merged**